From: Christoph Hellwig <hch@lst.de>
Date: Sat, 16 May 2020 20:27:59 +0200
Subject: [PATCH] blk-mq: remove a pointless queue enter pair in
 blk_mq_alloc_request
Git-commit: 22fa792cd8c67958adbddc9d436b850e148a623b
Patch-mainline: v5.8-rc1
References: bsc#1175995,jsc#SLE-15608

No need for two queue references.

Signed-off-by: Christoph Hellwig <hch@lst.de>
Reviewed-by: Bart Van Assche <bvanassche@acm.org>
Signed-off-by: Jens Axboe <axboe@kernel.dk>
Acked-by: Hannes Reinecke <hare@suse.com>
---
 block/blk-mq.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/block/blk-mq.c b/block/blk-mq.c
index b7e06673a30b..4e62b97dceb4 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -406,10 +406,7 @@ struct request *blk_mq_alloc_request(struct request_queue *q, unsigned int op,
 	if (ret)
 		return ERR_PTR(ret);
 
-	blk_queue_enter_live(q);
 	rq = blk_mq_get_request(q, NULL, &alloc_data);
-	blk_queue_exit(q);
-
 	if (!rq)
 		goto out_queue_exit;
 	rq->__data_len = 0;
-- 
2.16.4

